packages/office-ui-fabric-react/src/components/Button/Button.tsx

```diff
@@ -1,137 +1,48 @@
-  public static defaultProps: IButtonProps = {
-    buttonType: ButtonType.default
-  };
-  private _buttonElement: HTMLButtonElement;
-
-  constructor(props: IButtonProps) {
-    super(props, { 'rootProps': null });
-
-    this.state = {
-      labelId: getId('Button'),
-      descriptionId: getId('Button'),
-      ariaDescriptionId: getId('Button')
-    };
-  }
+import { BaseComponent } from '../../Utilities';
+import { ButtonType, IButton, IButtonProps } from './Button.Props';
+import { BaseButton } from './BaseButton';
 
-  public render(): JSX.Element {
-    let { buttonType, children, icon, description, ariaLabel, ariaDescription, href, disabled, onClick } = this.props;
-    let { labelId, descriptionId, ariaDescriptionId } = this.state;
+import { DefaultButton } from './DefaultButton/DefaultButton';
+import { CommandButton } from './CommandButton/CommandButton';
+import { CompoundButton } from './CompoundButton/CompoundButton';
+import { IconButton } from './IconButton/IconButton';
+import { PrimaryButton } from './PrimaryButton/PrimaryButton';
 
-    const renderAsAnchor: boolean = !!href;
-    const tag = renderAsAnchor ? 'a' : 'button';
-    const nativeProps = getNativeProps(this.props.rootProps || this.props, renderAsAnchor ? anchorProperties : buttonProperties);
-    const className = css((this.props.className), {
-      'ms-Button ms-Button--default': buttonType === ButtonType.default || buttonType === ButtonType.normal,
-      'ms-Button ms-Button--primary': buttonType === ButtonType.primary,
-      'ms-Button ms-Button--hero': buttonType === ButtonType.hero,
-      'ms-Button ms-Button--compound': buttonType === ButtonType.compound,
-      'ms-Button ms-Button--command': buttonType === ButtonType.command,
-      'ms-Button ms-Button--icon': buttonType === ButtonType.icon,
-      'disabled': (renderAsAnchor && disabled) // add disable styling if it is an anchor
-      // if not utilize default button disabled behavior.
-    });
+/**
+ * @deprecated
+ * This class is deprecated. Use the individual *Button components instead.
+ */
+export class Button extends BaseComponent<IButtonProps, {}> implements IButton {
+  private _button: BaseButton;
 
-    const iconSpan = icon && (buttonType === ButtonType.command || buttonType === ButtonType.hero || buttonType === ButtonType.icon)
-      ? <span className='ms-Button-icon'><i className={ `ms-Icon ms-Icon--${icon}` }></i></span>
-      : null;
+  public render() {
+    let props = this.props;
 
-    // ms-Button-description is only shown when the button type is compound.
-    // In other cases it will not be displayed.
-    const descriptionSpan: React.ReactElement<React.HTMLProps<HTMLSpanElement>> = description
-      ? <span className='ms-Button-description' id={ descriptionId }>{ description }</span>
-      : null;
+    switch (props.buttonType) {
+      case ButtonType.command:
+        return <CommandButton ref={ this._resolveRef('_button') } { ...props } />;
```

Comment:
I think the order of these might need to be reversed. Unless you want someone to be able to override your resolveRef.